VoxelSniperCore/src/main/java/com/github/kevindagame/util/VersionChecker.kt

```diff
@@ -0,0 +1,122 @@
+package com.github.kevindagame.util
+
+import com.google.gson.Gson
+import java.io.InputStreamReader
+import java.net.HttpURLConnection
+import java.net.URL
+
+/**
+ * @author KevinDaGame
+ * A class that checks for the latest version of VoxelSniper-Reimagined on GitHub.
+ */
+class VersionChecker() {
+    private val releasesApiUrl = "https://api.github.com/repos/$REPOSITORY_OWNER/$REPOSITORY_NAME/releases"
+
+    /**
+     * Gets the latest version info from the GitHub API.
+     * @param currentVersion The current version of VoxelSniper-Reimagined.
+     *
+     * @return The latest version info, or null if the current version is the latest.
+     */
+    fun getLatestVersionInfo(currentVersion: String): VersionInfo? {
+        val releases = getReleasesFromApi()
+        if (releases.isEmpty()) {
+            return null
+        }
+        val latestRelease = releases[0]
+        val latestVersion = latestRelease.tag_name.removePrefix("v")
+        if (isOutdated(currentVersion, latestVersion)) {
+            val downloadUrl = latestRelease.html_url
+            LATEST_VERSION = VersionInfo(latestVersion, downloadUrl)
+        }
+        return null
+    }
+
+    /**
+     * Gets the latest version info from the GitHub API.
+     * @return The latest version info as a list of releases.
+     */
+    private fun getReleasesFromApi(): List<Release> {
+        return try {
+            val conn = URL(releasesApiUrl).openConnection() as HttpURLConnection
+            conn.addRequestProperty("User-Agent", "Mozilla/5.0")
+            val reader = InputStreamReader(conn.inputStream)
+            val releases = Gson().fromJson(reader, Array<Release>::class.java)
+            reader.close()
+            conn.disconnect()
+            releases.toList()
+        } catch (e: Exception) {
+            System.err.println("Failed to check for a new version of VoxelSniper-Reimagined.")
+            emptyList()
+        }
+    }
+
+    /**
+     * Checks if the current version is outdated.
+     * @param currentVersion The current version of VoxelSniper-Reimagined.
+     * @param latestVersion The latest version of VoxelSniper-Reimagined.
+     *
+     * @return True if the current version is outdated, false otherwise.
+     */
+    private fun isOutdated(currentVersion: String, latestVersion: String): Boolean {
+        val current = parseVersion(currentVersion)
+        val latest = parseVersion(latestVersion)
+        return current < latest
+    }
+
+    /**
+     * Parses a version string into a Version object.
+     * @param version The version string.
+     *
+     * @return The parsed version.
+     */
+    private fun parseVersion(version: String): Version {
+        val versionParts = version.removePrefix("v").split(".")
```

Comment:
Might be good to make it lowercase before removing the prefix
```suggestion
        val versionParts = version.lowercase().removePrefix("v").split(".")
```

Reply:
Why did you remove this again?